core/src/main/java/com/linecorp/armeria/common/logging/DefaultTextLogFormatter.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.util.TextFormatter;
+import com.linecorp.armeria.internal.common.util.TemporaryThreadLocals;
+
+/**
+ * A formatter that convert {@link RequestLog} into text message.
+ */
+final class DefaultTextLogFormatter implements LogFormatter {
+
+    static final DefaultTextLogFormatter DEFAULT_INSTANCE = new DefaultTextLogFormatter();
+
+    @Override
+    public String formatRequest(RequestLog log, LogSanitizers<RequestHeaders> sanitizers) {
+        requireNonNull(log, "log,");
+        requireNonNull(sanitizers, "sanitizers,");
+
+        final Set<RequestLogProperty> availableProperties = log.availableProperties();
+        if (!availableProperties.contains(RequestLogProperty.REQUEST_START_TIME)) {
+            return "{}";
+        }
+
+        final String requestCauseString;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_CAUSE) && log.requestCause() != null) {
+            requestCauseString = String.valueOf(log.requestCause());
+        } else {
+            requestCauseString = null;
+        }
+
+        final String sanitizedHeaders;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_HEADERS) && log.requestHeaders() != null) {
```

Comment:
ditto for the json formatter

```suggestion
        if (availableProperties.contains(RequestLogProperty.REQUEST_HEADERS)) {
```